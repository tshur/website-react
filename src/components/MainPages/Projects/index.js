import React from 'react';
import './index.scss';

import Game from '../../../playground/tic-tac-toe/components/Game';

const Projects = () =>
  <div className='page-container projects'>
    <div className='container'>
      <h1> Projects </h1>
      <br />
      <Game />
    </div>
  </div>

export default Projects;