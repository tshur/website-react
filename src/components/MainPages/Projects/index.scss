.projects {
    .projects-body {
        margin-top: 100px;
        margin-left: 15%;
        margin-right: 15%;
        margin-bottom: 100px;

        .title {
            margin-left: 20px;
            text-transform: uppercase;
        }
        .project-grid {
            display: flex;
            flex-wrap: wrap;

            .project-card {
                display: flex;
                flex-direction: column;
                background: repeating-linear-gradient(
                    -45deg,
                    #ffffff,
                    #ffffff 30px,
                    #eeeeee 30px,
                    #eeeeee 60px
                );
                margin: 20px;
                padding: 40px;
                height: 300px;
                width: 200px;
                box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
                transition: all 0.3s cubic-bezier(.25,.8,.25,1);

                &:hover {
                    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
                }

                .coming-soon {
                    text-align: center;
                    margin-top: auto;
                }
            }
        }
    }
}