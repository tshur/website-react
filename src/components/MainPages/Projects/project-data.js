const projectData = [
    {
        'name': 'Tic-Tac-Toe',
        'link': '',
        'desc': 'A simple tic-tac-toe game built in React following the official React tutorial.'
    },
    {
        'name': 'Calculator',
        'link': '',
        'desc': 'An calculator app built in React based on the iOS phone calculator.'
    },
    {
        'name': 'Reddit Feed',
        'link': '',
        'desc': 'An Reddit feed clone built in React. Uses the Reddit API to grab posts from a given subreddit and display them as a list of cards.'
    },
    {
        'name': 'Questioner',
        'link': '',
        'desc': 'Web app to help the user brainstorm by asking questions and recording responses. Senior design project (in progress) with Robbie Culkin.'
    },
];

export default projectData;
