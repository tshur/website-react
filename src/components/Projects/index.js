import React from 'react';
import styles from './index.css';

import Feed from '../../playground/reddit-feed/components/Feed';

const Projects = () =>
  <div className={`container ${styles.projects}`}>
    <h1> Projects </h1>
    <br />
    <Feed />
  </div>

export default Projects;