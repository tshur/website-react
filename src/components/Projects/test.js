import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Projects from '.';

describe('Projects', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Projects />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Projects />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
