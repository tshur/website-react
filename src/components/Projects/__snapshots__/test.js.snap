// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects has a valid snapshot 1`] = `
<div
  className="container undefined"
>
  <h1>
     Projects 
  </h1>
  <br />
  <div
    className={undefined}
  >
    <h1>
      Simple Calculator
    </h1>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        0
      </div>
    </div>
    <div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            C
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            +/-
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            %
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            /
          </button>
        </div>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            7
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            8
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            9
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            *
          </button>
        </div>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            4
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            5
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            6
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            -
          </button>
        </div>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            1
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            2
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            3
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            +
          </button>
        </div>
      </div>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            0
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            .
          </button>
        </div>
        <div
          className={undefined}
        >
          <button
            className="sc-bdVaJa dyCkzW"
            onClick={[Function]}
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
